contrib/parseq-retry/src/main/java/com/linkedin/parseq/retry/AbstractRetryPolicy.java

```diff
@@ -0,0 +1,146 @@
+package com.linkedin.parseq.retry;
+
+import com.linkedin.parseq.retry.backoff.BackoffPolicy;
+import com.linkedin.parseq.retry.monitor.EventMonitor;
+import com.linkedin.parseq.retry.termination.TerminationPolicy;
+
+import java.util.function.Function;
+
+
+/**
+ * A base class for policies that enable customizable retries for arbitrary parseq tasks.
+ *
+ * @param <T> Type of a task result, used for strongly typed processing of outcomes.
+ *
+ * @author Oleg Anashkin (oleg.anashkin@gmail.com)
+ */
+public abstract class AbstractRetryPolicy<T> {
```

Comment:
@angxu What do you think about merging `RetryPolicy` and `AbstractRetryPolicy`? I don't foresee any use cases for extending either of them, so having only one class `RetryPolicy` as both builder and container simplifies things.